### DISCUSSION TITLE
Feature Request: Provide variables map for alert message templates

### DISCUSSION THREAD

**Discussion by stefankaufmann** [grafana/grafana#55144] [Alerting]
Hi, 
we recently upgraded our grafana from version 8.5 to 9.1. The Teams alert messages now contain a lot of unwanted information.

Message templating is quite cumbersome since the Alert object only holds a field `ValueString` which is hard to parse. However, when adding annotations in the alert rule itself, one can use the `$values` variable to get the same information as a `map` which can be used for iterating. 
Thanks to [this ](https://community.grafana.com/t/how-to-use-alert-message-templates-in-grafana/67537/3) great answer, we managed to add the information we want as an annotation. 

**The problem** with this solution is, that we need to adjust all alert rules and to add the annotation. It would be much nicer, if the `$values` map was accessible also in the message templating.  

**Example:** 
We have an alert configured like this: 
![image](https://user-images.githubusercontent.com/7921443/190079059-22ae6625-184d-48c3-a4c3-7264e83bfec5.png)

The information we want to have in the alert is: All the alerting `container.name` field and the corresponding alert value. Here is an example of what we achieved with the process explained above: 

![image](https://user-images.githubusercontent.com/7921443/190080384-5e244c4e-86fc-49b9-b119-4d7e81b24c52.png)

We achieved this by using the following annotation in the alert rule
![image](https://user-images.githubusercontent.com/7921443/190080659-d0f230a2-5f37-400e-a286-69c247347adc.png)
Here is the code again to copy/paste
```
{{ with $values }}{{ range $k, $v := . }}{{ range $k1, $v1 := $v.Labels }}
 - **{{ $v1 }}** [{{ $v }}]
{{ end }}{{ end }}{{ end }}`
```

And the following message template 
```
{{ define "mymessage" }}
{{ range .Alerts }}
{{ .Annotations.AlertValues }}    
Panel: [{{ .PanelURL }}]({{ .PanelURL }})
{{ end }}
{{ end }}
```

Although this setup works, we need to add the annotation code to each alert rule, which is not easy to maintain

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
